src/core/webview/ClineProvider.ts

```diff
@@ -2458,7 +2493,8 @@ export class ClineProvider extends EventEmitter<ClineProviderEvents> implements
 	}
 
 	async showTaskWithId(id: string) {
-		if (id !== this.getCurrentCline()?.taskId) {
+		const currentCline = await this.getCurrentCline()
```

Comment:
There are loads of changes like this, which seem to be necessitated by the fact that getCurrentCline thas become async.  But I don't understand what is driving that complexity.